olive/evaluator/olive_evaluator.py

```diff
@@ -1058,16 +1075,24 @@ def _prepare_dataloader(
         return FileListCommonDataLoader(dataloader, model.io_config, batch_size=file_chunk_size)
 
 
-class OliveEvaluatorFactory:
-    @staticmethod
-    def create_evaluator_for_model(model: "OliveModelHandler") -> OliveEvaluator:
-        evaluator_cls = OliveEvaluator.registry[str(model.framework).lower()]
-        return evaluator_cls()
+class OliveEvaluatorConfig(NestedConfig):
+    _nested_field_name = "type_args"
+
+    type: str = None
+    type_args: Dict = Field(default_factory=dict)
 
+    # user script to define and register the evaluator
+    user_script: Union[Path, str] = None
+    script_dir: Union[Path, str] = None
 
-class OliveEvaluatorConfig(ConfigBase):
     metrics: List[Metric] = []  # noqa: RUF012
```

Comment:
I am not really sure how the custom evaluator would work since it still needs metric. Is the idea that the metric for the custom evaluator (unless they are latency or accuracy) would be defined as `custom` metric type? 

In that case, there is overlap between the custom metric type's options https://github.com/microsoft/Olive/blob/64b79dcddd029ae2927637bb03074b46958763e2/olive/evaluator/metric_config.py#L29 and https://github.com/microsoft/Olive/blob/64b79dcddd029ae2927637bb03074b46958763e2/olive/evaluator/metric_config.py#L18 
The custom metric is also able to take an `eval_func` which can be run directly on the model https://github.com/microsoft/Olive/blob/64b79dcddd029ae2927637bb03074b46958763e2/olive/evaluator/olive_evaluator.py#L145

can you share an example or test for how such an evaluator config would be defined and used?

Reply:
just saw your comment about custom metric deprecation. If that addresses this question, then the PR looks good to me.